src/lib.rs

```diff
@@ -380,6 +380,7 @@ pub use self::module::{ModuleInstance, ModuleRef, ExternVal, NotStartedModuleRef
 pub use self::global::{GlobalInstance, GlobalRef};
 pub use self::func::{FuncInstance, FuncRef, FuncInvocation, ResumableError};
 pub use self::types::{Signature, ValueType, GlobalDescriptor, TableDescriptor, MemoryDescriptor};
+pub use self::validation::{ValidatedModule, validate_module};
```

Comment:
This will expose `ValidatedModule` which in turn will ultimately expose `isa::Instruction`, which is the implementation detail.